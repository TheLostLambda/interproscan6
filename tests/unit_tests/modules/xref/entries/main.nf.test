nextflow_process {

    name "Test Process ENTRIES"
    script "interproscan/modules/xrefs/entries/main.nf"
    process "ENTRIES"

    stage {
        copy "tests/unit_tests/test_inputs/xrefs/entries.json"
        copy "interproscan/scripts/xrefs/entries.py"
        copy "tests/unit_tests/test_inputs/xrefs/subworkflow/xref_matches_input.json"
    }

    test("Should run without failures") {

        when {
            params {
            }
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/xrefs/subworkflow/xref_matches_input.json")
                input[1] = file("meta/tests/unit_tests/test_inputs/xrefs/entries.json")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
