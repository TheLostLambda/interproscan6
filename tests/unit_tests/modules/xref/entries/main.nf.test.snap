{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "entries_xref_matches_input.json:md5,c46feebfeb2119a9982489815045fd82"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-09T09:11:14.801795"
    }
}